var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -52,13 +52,10 @@ def configure_args(self):
         if spec.satisfies('%cce@9:'):
             args.append('LDFLAGS=-rtlib=compiler-rt')
 
-        if spec.satisfies('%clang') and not spec.satisfies('platform=darwin'):
-            args.append('LDFLAGS=-rtlib=compiler-rt')
-
-        if spec.satisfies('%arm') and not spec.satisfies('platform=darwin'):
-            args.append('LDFLAGS=-rtlib=compiler-rt')
-
-        if spec.satisfies('%fj') and not spec.satisfies('platform=darwin'):
+        if spec.satisfies('%clang') or \
+           spec.satisfies('%aocc') or \
+           spec.satisfies('%arm') or \
+           spec.satisfies('%fj') and not spec.satisfies('platform=darwin'):
```

Comment:
`and` has a higher precedence than `or`: https://docs.python.org/3/reference/expressions.html#boolean-operations

I think you'll need parentheses around the or expressions.

Reply:
Added parentheses around the or expressions.